sendgrid/base_interface.py

```diff
@@ -60,3 +65,24 @@ def send(self, message):
             message = message.get()
 
         return self.client.mail.send.post(request_body=message)
+
+    def set_host(self,host):
+        self.host = host
+
+    def set_region(self,region):
```

Comment:
Rather than calling it set_region, should we call it "set_data_residency()", as per the [doc](https://docs.google.com/document/d/130NIuBI1XnwaCTJPsHSNQEgw_9K7SthZD6EjT2PeB-E/edit) ?